src/core/server/saved_objects/migrationsv2/model.test.ts

```diff
@@ -1016,18 +1086,139 @@ describe('migrations v2 model', () => {
+            corruptDocumentIds: [...corruptDocumentIds],
+            transformErrors: [],
+          };
+          const res: ResponseType<'OUTDATED_DOCUMENTS_TRANSFORM'> = Either.right({ processedDocs });
+          const newState = model(
+            outdatedDocumentsTransformStateWithFailedDocuments,
+            res
+          ) as OutdatedDocumentsSearchRead;
+          expect(newState.controlState).toEqual('OUTDATED_DOCUMENTS_SEARCH_READ');
+          expect(newState.corruptDocumentIds).toEqual(corruptDocumentIds);
+          expect(newState.retryCount).toEqual(0);
+          expect(newState.retryDelay).toEqual(0);
+        });
+        test('OUTDATED_DOCUMENTS_TRANSFORM -> OUTDATED_DOCUMENTS_SEARCH_READ if there are are existing documents that failed transformation because of transform errors', () => {
+          const outdatedDocumentsTransformStateWithFailedDocuments: OutdatedDocumentsTransform = {
+            ...outdatedDocumentsTransformState,
+            corruptDocumentIds: [],
+            transformErrors: [...transformationErrors],
+          };
+          const res: ResponseType<'OUTDATED_DOCUMENTS_TRANSFORM'> = Either.right({ processedDocs });
+          const newState = model(
+            outdatedDocumentsTransformStateWithFailedDocuments,
+            res
+          ) as OutdatedDocumentsSearchRead;
+          expect(newState.controlState).toEqual('OUTDATED_DOCUMENTS_SEARCH_READ');
+          expect(newState.corruptDocumentIds.length).toEqual(0);
+          expect(newState.transformErrors.length).toEqual(1);
+          expect(newState.retryCount).toEqual(0);
+          expect(newState.retryDelay).toEqual(0);
+        });
+      });
+      describe('OUTDATED_DOCUMENTS_TRANSFORM if action fails', () => {
+        test('OUTDATED_DOCUMENTS_TRANSFORM -> OUTDATED_DOCUMENTS_SEARCH_READ adding newly failed documents to state if documents failed the transform', () => {
+          const corruptDocumentIds = ([Symbol('raw saved object doc ID')] as unknown) as string[];
+          const res: ResponseType<'OUTDATED_DOCUMENTS_TRANSFORM'> = Either.left({
+            type: 'documents_transform_failed',
+            corruptDocumentIds,
+            transformErrors: [],
+          });
+          const newState = model(
+            outdatedDocumentsTransformState,
+            res
+          ) as OutdatedDocumentsSearchRead;
+          expect(newState.controlState).toEqual('OUTDATED_DOCUMENTS_SEARCH_READ');
+          expect(newState.corruptDocumentIds).toEqual(corruptDocumentIds);
+        });
+        test('OUTDATED_DOCUMENTS_TRANSFORM -> OUTDATED_DOCUMENTS_SEARCH_READ combines newly failed documents with those already on state if documents failed the transform', () => {
+          const corruptDocumentIds = ([Symbol('raw saved object doc ID')] as unknown) as string[];
+          const transformationErrors = ([
+            {
+              rawId: Symbol('transform error object') as unknown,
```

Comment:
why we need all these `as unknown`? can we get rid of them?

Reply:
Done